specification/sections/03-evaluation-context.md

```diff
@@ -128,4 +132,82 @@ The SDK implementation must run the `on context changed` handler on all register
 
-The SDK implementation must run the `on context changed` handler only on the provider that is scoped to the mutated `evaluation context`.
\ No newline at end of file
+The SDK implementation must run the `on context changed` handler only on the provider that is scoped to the mutated `evaluation context`.
+
+### 3.3 Context Propagation
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+`Transaction context` is a container for transaction-specific `evaluation context` (e.g. user id, user agent, IP).
+Transaction context can be set where specific data is available (e.g. an auth service or request handler) and by using the `transaction context propagator` it will automatically be applied to all flag evaluations within a transaction (e.g. a request or thread).
+
+The following shows a possible TypeScript implementation using [AsyncLocalStorage (async_hooks)](https://nodejs.org/api/async_context.html):
+
+```typescript
+export class AsyncLocalStorageTransactionContext implements TransactionContextPropagator {
+    private asyncLocalStorage = new AsyncLocalStorage<EvaluationContext>();
+
+    getTransactionContext(): EvaluationContext {
+        return this.asyncLocalStorage.getStore() ?? {};
+    }
+    setTransactionContext(context: EvaluationContext, callback: () => void): void {
+        this.asyncLocalStorage.run(context, callback);
+    }
+}
+
+/**
+ * This example is based on an express middleware.
+ */
+app.use((req: Request, res: Response, next: NextFunction) => {
+    const ip = res.headers.get("X-Forwarded-For")
+    OpenFeature.setTransactionContext({ targetingKey: req.user.id, ipAddress: ip }, () => {
+        // The transaction context is used in any flag evaluation throughout the whole call chain of next 
+        next();
+    });
+})
+```
+
+#### Condition 3.3.1
+
+> The implementation uses the dynamic-context paradigm.
+
+see: [dynamic-context paradigm](../glossary.md#dynamic-context-paradigm)
+
+##### Conditional Requirement 3.3.1.1
+
+> The API **MUST** have a method for setting a `transaction context propagator`.
+
+If there already is a `transaction context propagator`, it is replaced with the new one. 
+
+##### Conditional Requirement 3.3.1.2
+
+> The API **MUST** have a method for setting the `evaluation context` of the `transaction context propagator` for the current transaction.
+
+If a `transaction context propagator` is set, the SDK will call the [method defined in 3.3.1.3](#conditional-requirement-3313) with this `evaluation context` and so this `evaluation context` will be available during the current transaction.
+If no `transaction context propagator` is set, this `evaluation context` is not used for evaluations.
+This method can e.g. be used in a request handler to add request-specific information to the `evaluation context`.
```

Comment:
```suggestion
This method can be used in a request handler to add request-specific information to the `evaluation context`.
```

Reply:
Without the e.g it feels like request handlers are the only intended use case to me @Kavindu-Dodan, but I am not sure how to state it differently.

Reply:
ohh I thought this was a mistake. It was odd to see e.g in a sentence, may be we can expand this to something like 

```suggestion
This method then can be used for example in a request handler to add request-specific information to the `evaluation context`.
```